### PR TITLE
Use Static Functions

### PR DESCRIPTION
**Pull request by stephencelis**
Here's a preliminary PR for what we talked about!

Right now I use `attributes: […]` consistently everywhere, but functions with no content could use `_ attributes: Attribute...` instead.

``` swift
.link(attributes: [.href("blob"), .rel(.alternate)])
// vs.
.link(.href("blob"), .rel(.alternate))
```

We could also add overloads to accept `.async` wherever booleans like `.async(true)` work.

### CODE REVIEW

File: Html.playground/Contents.swift [hunk 1]
```diff
@@ -32,31 +32,30 @@ code {
...
 /// A document built in the HTML DSL.
-let doc = document(
-  doctype,
-  html(
-    head(
-      style(unsafe: stylesheet)
+let doc: Node = .document(
+  .html(
+    .head(
+      .style(safe: stylesheet)
```
Comment (stephencelis) [inline Html.playground/Contents.swift:38]: One weird concession was without making `safe` vs. `unsafe` explicit, I'd get ambiguous errors that required `as StaticString` even on string literals. Was strange and maybe worth filing a bug for!

File: Tests/HtmlSnapshotTestingTests/HtmlSnapshotTestingTests.swift [hunk 2]
```diff
@@ -9,313 +9,313 @@ typealias SnapshotTestCase = XCTestCase
...
-            [Html.class("bg-purple150 d-pt4 d-pb4 m-pl2 m-pr2 m-pt2 m-pb2")],
-            nav(
-              div(
-                [Html.class("col-m")],
-                a(
-                  [href("/")],
-                  img(base64: "ZnVuY3Rpb25z", type: .image(.svg), alt: "", [Html.class("hide-d")])
+        .body(
+          .header(
+            attributes: [.class("bg-purple150 d-pt4 d-pb4 m-pl2 m-pr2 m-pt2 m-pb2")],
```
Comment (stephencelis) [inline Tests/HtmlSnapshotTestingTests/HtmlSnapshotTestingTests.swift:71]: `.class` is feelin nicer than `Html.class` or backticks!

File: Html.playground/Contents.swift (hunk 1 above)
Comment (mbrandonw) [reply Html.playground/Contents.swift:38]: oh dang strange

File: Tests/HtmlSnapshotTestingTests/__Snapshots__/HtmlSnapshotTestingTests/testSnapshots.1.html [hunk 3]
```diff
@@ -1,136 +1,79 @@
 <a>
-  foo1
```
Comment (mbrandonw) [inline Tests/HtmlSnapshotTestingTests/__Snapshots__/HtmlSnapshotTestingTests/testSnapshots.1.html:2]: are all the deletions in this file intended?

**Review by mbrandonw — APPROVED** [review]
niiiice

File: Tests/HtmlSnapshotTestingTests/__Snapshots__/HtmlSnapshotTestingTests/testSnapshots.1.html (hunk 3 above)
Comment (stephencelis) [reply Tests/HtmlSnapshotTestingTests/__Snapshots__/HtmlSnapshotTestingTests/testSnapshots.1.html:2]: I think so but will audit. We were testing each overload before.

**Comment by stephencelis** [comment]
Gonna merge for now! Can make future adjustments before any official release.

**Pull request merged**